### ISSUE TITLE
Error in "DNNTransferLearning" solution

### ISSUE THREAD

**Issue by l1pus** [dotnet/machinelearning-samples#677]
Hey,

I`m having problem with starting sample solution indicated in subject of this issue.
While running the start project, all is fine and dandy untill command :

`       var pipeline = mlContext.Transforms.Conversion.MapValueToKey(outputColumnName: "LabelAsKey",
                                                                            inputColumnName: "Label",
                                                                            keyOrdinality: ValueToKeyMappingEstimator.KeyOrdinality.ByValue)
                        .Append(mlContext.Model.ImageClassification("ImagePath", "LabelAsKey",
                                        arch: ImageClassificationEstimator.Architecture.ResnetV2101,
                                        epoch: 100,     //An epoch is one learning cycle where the learner sees the whole training data set.
                                        batchSize: 30,  // batchSize sets the number of images to feed the model at a time. It needs to divide the training set evenly or the remaining part won't be used for training.                              
                                        metricsCallback: (metrics) => Console.WriteLine(metrics),
                                        //OPTIONAL (*1*) 
                                        validationSet: transformedValidationDataView));`

Which results with exception : 

System.FormatException: „Tensorflow exception triggered while loading model.”
InvalidProtocolBufferException: While parsing a protocol message, the input ended unexpectedly in the middle of a field.  This could mean either that the input has been truncated or that an embedded message misreported its own length.

I have tried InceptionV3 instead od Resnet , but exception is shown a bit later in the code...

**Comment by CESARDELATORRE** [comment]
I was not able to repro this issue. Must be related to base software in the OS?

In any case, we just updated this feature to Preview-2 which has great improvements such as:

Can score with in-memory images
Can use GPU
I just updated the same sample (still not using GPU in the sample, though, since it needs to use a different NuGet package and you need to install CUDA, too), but can you try this new version of the sample? In the tests we're doing it works out-of-the-box.

Read the Samples's README.MD if you want so see the changed highlighted compared to Preview1. Most of the changes are related to the in-memory images usage.

Confirm if you still have the same issue so we can research what's about. Probably any missing dependency out of ML.NET / .NET.

Thanks,

**Comment by luisquintanilla** [comment]
Closing this issue since a response was provided and there has been no additional activity. Also, the ImageClassification API is not generally available in the stable version of ML.NET.

https://docs.microsoft.com/en-us/dotnet/machine-learning/tutorials/image-classification-api-transfer-learning
